src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: T): GroupsMap<T> {
```

Comment:
The type of the 'key' parameter should be 'keyof T' instead of 'T'. This ensures that the key parameter must be one of the keys of the generic type T.